x/metadata/keeper/scope_test.go

```diff
@@ -1370,6 +1401,41 @@ func (s *ScopeKeeperTestSuite) TestValidateScopeUpdateOwners() {
 			signers:  []string{s.user1, s.user2},
 			errorMsg: "",
 		},
+		{
+			name:     "smart contract without provenance role added",
+			existing: scopeWithOwners(ownerPartyList(s.user1)),
+			proposed: scopeWithOwners(ownerPartyList(s.user1, s.scUser)),
+			signers:  []string{s.user1},
+			errorMsg: `account "` + s.scUser + `" is a smart contract but does not have the PROVENANCE role`,
+		},
+		{
+			name:     "smart contract without provenance role removed",
+			existing: scopeWithOwners(ownerPartyList(s.user1, s.scUser)),
+			proposed: scopeWithOwners(ownerPartyList(s.user1)),
+			signers:  []string{s.scUser, s.user1},
+			errorMsg: "",
+		},
+		{
+			name:     "smart contract without provenance role removed but wrong signer order",
+			existing: scopeWithOwners(ownerPartyList(s.user1, s.scUser)),
+			proposed: scopeWithOwners(ownerPartyList(s.user1)),
+			signers:  []string{s.user1, s.scUser},
+			errorMsg: "smart contract signer " + s.scUser + " cannot follow non-smart-contract signer",
+		},
+		{
+			name:     "with rollup smart contract without provenance role added",
+			existing: rollupScopeWithOwners(pt(s.user1, owner, false)),
+			proposed: rollupScopeWithOwners(pt(s.user1, owner, false), pt(s.scUser, owner, true)),
+			signers:  []string{s.user1},
+			errorMsg: `account "` + s.scUser + `" is a smart contract but does not have the PROVENANCE role`,
+		},
+		{
+			name:     "with rollup smart contract without provenance role removed",
```

Comment:
Same as above.

Reply:
Same here. Only the proposed should matter for the provenance/smart-contract check.

This one threw for a second too, though. I thought it might need to be signed by the smart contract too, but the smart contract is optional, so it's not required as a signer.